File: english-hub-frontend/src/services/deckService.ts
import {
  DeckCreateRequest,
  DeckResponse,
  DeckUpdateRequest,
} from '@/types/deckType';
import axiosInstance from './axiosInstance';

export async function getAllDecks(): Promise<DeckResponse[]> {
  const response = await axiosInstance.get('/decks');
  return response.data.result;
}

export async function getDeckById(deckId: string): Promise<DeckResponse> {
  const response = await axiosInstance.get(`/decks/${deckId}`);
  return response.data.result;
}

export async function createDeck(
  deckData: DeckCreateRequest
): Promise<DeckResponse> {
  const response = await axiosInstance.post('/decks', deckData);
  return response.data.result;
}

export async function updateDeck(
  deckId: string,
  deckData: DeckUpdateRequest
): Promise<DeckResponse> {
  const response = await axiosInstance.put(`/decks/${deckId}`, deckData);
  return response.data.result;
}

export async function deleteDeck(deckId: string): Promise<string> {
  const response = await axiosInstance.delete(`/decks/${deckId}`);
  return response.data.message;
}

File: english-hub-backend/src/main/java/com/example/englishhubbackend/service/DeckService.java
package com.example.englishhubbackend.service;

import com.example.englishhubbackend.dto.request.DeckCreateRequest;
import com.example.englishhubbackend.dto.request.DeckUpdateRequest;
import com.example.englishhubbackend.dto.response.DeckResponse;
import java.util.List;
import java.util.UUID;

public interface DeckService {
  DeckResponse createDeck(DeckCreateRequest deckCreateRequest);

  List<DeckResponse> getAllDecks();

  DeckResponse getDeckById(UUID id);

  DeckResponse updateDeck(UUID id, DeckUpdateRequest deckUpdateRequest);

  void deleteDeck(UUID id);
}
